plaidml2/op/lib/ops.cc

```diff
@@ -1957,17 +1957,49 @@ Value reshape(const Value& value) {
   if (args.size() != 2) {
     throw std::runtime_error(str(boost::format("PlaidML reshape op expects 2 arguments (received %1%)") % args.size()));
   }
+
   auto I = args[0].as_tensor();
   std::vector<TensorDim> dims;
+  int64_t dim_counter = 0;
+  int64_t neg_idx = -1;
+  std::vector<TensorDim> I_dims(I.shape().ndims());
+  I.bind_dims(I_dims);
+
   for (auto dim : args[1].as_tuple()) {
     if (dim.is_int()) {
       dims.emplace_back(dim.as_int());
     } else if (dim.is_dim()) {
       dims.emplace_back(dim.as_dim());
     } else if (dim.is_str()) {
       // TODO: handle special cases
-      dims.emplace_back(0);
+      // TODO: figure out why there are quotation marks around fill and match
+      if (strcmp("\"match\"", dim.str().c_str()) == 0) {
+        dims.emplace_back(I_dims[dim_counter]);
+      } else if (strcmp("\"fill\"", dim.str().c_str()) == 0) {
+        if (neg_idx > -1) {
+          throw std::runtime_error(
+              str(boost::format("PlaidML reshape op - at most one dimension of size -1 may be provided")));
+        }
+        neg_idx = dim_counter;
+        dims.emplace_back(1);
+      }
+    } else if (dim.is_none()) {
+      dims.emplace_back(I_dims[dim_counter]);
     }
+    dim_counter++;
+  }
+
+  if (neg_idx > -1) {  // there was a -1 dimension which needs to be filled
```

Comment:
This will not work correctly if any of the relevant `I_dims` are symbolic (which doesn't get tested by `backend_test` as it currently exists). In the long run we need to think about how to test this best and implement that. But for now, see if changing num to a `TensorDim` type still works for the things we are testing, and if so do that.

Reply:
I just remembered that `opTest` has a parameter `input_shapes` used for this purpose. See `testConcatenateSymbolic` for an example of how to use it. We could add using it to test symbolic reshaping as a separate PR or as part of this PR